### DISCUSSION TITLE
How to get transcription and translation to be for same segment of audio?

### DISCUSSION THREAD

**Discussion by Kevin-French** [openai/whisper#1438] [Q&A] [answered]
Is there a way to get the translations and transcriptions to be for all the same time segments for a given audio?  I have tried to transcribe and translate the same audio file and get different periods of times for the segments for each task, also there are often different numbers of segments for the tasks.

**Comment by phineas-pta** [comment]
not with whisper

you must transcribe, then put transcript into translation

**Reply by Kevin-French** [reply]
By "then put transcript into translation" do you mean using a service that is not Whisper? Or can Whisper translate just text somehow too?

**Reply by phineas-pta** [reply, marked as answer]
whisper cannot do that

use other services / models

**Reply by Kevin-French** [reply]
@phineas-pta Ok, thank you

**Comment by Jawschamp** [comment]
This unfortunately isn't really the main goal of Whisper, as nice of a feature as this would be, that's gonna be a you issue, however I wouldn't imagine you'd have to much trouble with the translation of segments as there are tools that can identify parts of a string that has has a diffrent languages, which obviously you can deal with from there.

**Reply by Kevin-French** [reply]
@Jawschamp Do you think there would be a significant reduction in the quality of the translations if I used the start and end times from the segments from the transcription to break up the audio file and then give those broken up files to Whisper to translate?

**Reply by phineas-pta** [reply]
that depends on languages + how you break transcript
